app/src/controllers/submission.ts

```diff
@@ -443,18 +468,68 @@ const controller = {
     }
   },
 
+  submitDraft: async (req: Request<never, never, SubmissionIntake>, res: Response, next: NextFunction) => {
+    try {
+      const update = req.body.activityId && req.body.submissionId;
+
+      const { submission, appliedPermits, investigatePermits } = await controller.generateSubmissionData(
+        req,
+        IntakeStatus.SUBMITTED
+      );
+
+      let result;
+
+      if (update) {
+        // Update submission
+        result = await submissionService.updateSubmission({
+          ...submission,
+          ...generateUpdateStamps(req.currentContext)
+        });
+      } else {
+        // Create new submission
+        result = await submissionService.createSubmission({
+          ...submission,
+          ...generateCreateStamps(req.currentContext)
+        });
+      }
+
+      // Remove already existing permits for this activity
+      await permitService.deletePermitsByActivity(submission.activityId);
+
+      // Create each permit
+      await Promise.all(appliedPermits.map(async (x: Permit) => await permitService.createPermit(x)));
+      await Promise.all(investigatePermits.map(async (x: Permit) => await permitService.createPermit(x)));
```

Comment:
Concurrency: Not entirely sure if we should have the inner map callback function have an async and await in there, as that may cause issues with promise propagation back to the upper Promise.all. Might be able to have the arrow function only without the concurrent decorators.